internal/bootstrap/deployer_test.go

```diff
@@ -136,23 +133,32 @@ func (s *deployerSuite) TestDeployLocalCharmThatDoesNotExist(c *gc.C) {
 func (s *deployerSuite) TestDeployLocalCharm(c *gc.C) {
 	defer s.setupMocks(c).Finish()
 
-	// When deploying the local charm, ensure that we get the returned charm URL
-	// and the origin with the correct architecture and `/stable` suffix
-	// on the origin channel.
-
 	cfg := s.newConfig(c)
-	s.ensureControllerCharm(c, cfg.DataDir)
+	archivePath, archiveSize := s.ensureControllerCharm(c, cfg.DataDir)
 
-	s.expectLocalCharmUpload(c)
+	hash := "edd4580b1d913ff312aa842b8be3ed743cc04c14633b51fb55db6332a299ae07"
+
+	s.applicationService.EXPECT().ResolveControllerCharmDownload(gomock.Any(), domainapplication.ResolveControllerCharmDownload{
+		SHA256: hash,
+		SHA384: "8f108ec3b0b858c07cb36c8ef7738682f3914673f31f775867af7679d37b7c4d65fed1adb5a04dc4f0aae85f9db6a062",
+		Path:   archivePath,
+		Size:   archiveSize,
+	}).Return(domainapplication.ResolvedControllerCharmDownload{
+		Charm:           s.charm,
+		ArchivePath:     "path",
+		ObjectStoreUUID: "1234",
+	}, nil)
 
 	deployer := s.newBaseDeployer(c, cfg)
 
 	info, err := deployer.DeployLocalCharm(context.Background(), "arm64", base.MakeDefaultBase("ubuntu", "22.04"))
 	c.Assert(err, jc.ErrorIsNil)
-	c.Assert(info.URL.String(), gc.Equals, "local:arm64/juju-controller-0")
+	c.Assert(info.URL.String(), gc.Equals, "local:juju-controller-0")
```

Comment:
Why don't we need to worry about this?

Reply:
We no longer store the URL in the database like this. This could be a string full of garbage. I'm trying to work out if we need it at all.